backend/app/auth_handlers.go

```diff
@@ -0,0 +1,246 @@
+
+}
+
+func (a *App) SignIn(ctx *gin.Context) (interface{}, Response) {
+
+	var requestBody models.User
+	err := json.NewDecoder(ctx.Request.Body).Decode(&requestBody)
+	if err != nil {
+
+		return nil, BadRequest(err)
+	}
+
+	var user models.User
+	result := a.client.Client.First(&user, "Email = ?", requestBody.Email)
+
+	if result.Error != nil {
+		if result.Error == gorm.ErrRecordNotFound {
+			return nil, NotFound(errors.New("invalid email or password"))
+		}
+		return nil, InternalServerError(result.Error)
+	}
+
+	isPasswordMatches := internal.CheckPasswordHash(requestBody.Password, user.Password)
+
+	if !isPasswordMatches {
+
+		return nil, NotFound(errors.New("invalid email or password"))
+	}
+
+	if !user.Verified {
+		return nil, Forbidden(errors.New("account not verified"))
+	}
+	// generate tokens
+	accessToken, err := internal.GenerateAccessToken(user)
+
+	if err != nil {
+		return nil, InternalServerError(err)
+	}
+
+	refreshToken, err := internal.GenerateRefreshToken(user)
+
+	if err != nil {
+		return nil, InternalServerError(err)
+	}
+
+	user = models.User{
+		Name:     user.Name,
+		Email:    user.Email,
+		Verified: user.Verified,
+	}
+
+	ctx.SetSameSite(http.SameSiteLaxMode)
+	ctx.SetCookie("Authorization", accessToken, 60*15, "", "", true, true)
+
+	return ResponseMsg{Message: "logged in successfully", Data: struct {
+		AccessToken  string      `json:"access_token"`
+		RefreshToken string      `json:"refresh_token"`
+		User         models.User `json:"user"`
+	}{
+		AccessToken:  accessToken,
+		RefreshToken: refreshToken,
+		User:         user,
+	}}, Ok()
+}
+
+func (a *App) UpdateUser(ctx *gin.Context) (interface{}, Response) {
+	userPayload, exists := ctx.Get("user")
+
+	if !exists {
+		return nil, UnAuthorized(errors.New("user not found"))
+	}
+
+	var user models.User
+
+	if err := json.Unmarshal([]byte(userPayload.(string)), &user); err != nil {
+		return nil, InternalServerError(err)
+	}
+
+	_, err := a.client.GetUserByID(user.ID)
+	if err != nil {
+		return nil, NotFound(errors.New("user not found"))
+	}
+
+	userID := user.ID
+	err = json.NewDecoder(ctx.Request.Body).Decode(&user)
+	if err != nil {
+		return nil, BadRequest(err)
+	}
+
+	user.ID = userID
+
+	err = a.client.UpdateUser(user)
+
+	if err != nil {
+
+		return nil, InternalServerError(err)
+	}
+
+	return ResponseMsg{Message: "updated successfully"}, Created()
+}
+
+func (a *App) GetUser(ctx *gin.Context) (interface{}, Response) {
+
+	userPayload, exists := ctx.Get("user")
+
+	if !exists {
+		return nil, NotFound(errors.New("user not found"))
+	}
+
+	var user models.User
+
+	if err := json.Unmarshal([]byte(userPayload.(string)), &user); err != nil {
+		return nil, InternalServerError(err)
+	}
+
+	user, err := a.client.GetUserByID(user.ID)
+
+	if err != nil {
+		return nil, NotFound(errors.New("user not found"))
+	}
+
+	return ResponseMsg{Message: "found", Data: user}, Ok()
+}
+
+func (a *App) RefreshToken(ctx *gin.Context) (interface{}, Response) {
```

Comment:
you should take the access token from your Authorization in headers and it will be a bearer token